src/complex/Filter/FilterList.cpp

```diff
@@ -73,6 +69,18 @@ AbstractPlugin* FilterList::getPlugin(const FilterHandle& handle) const
   return nullptr;
 }
 
+AbstractPlugin* FilterList::getPlugin(const std::string& humanOrClassName) const
+{
+  for(const auto& handle : getFilterHandles())
+  {
+    if(handle.getFilterName() == humanOrClassName || handle.getClassName() == humanOrClassName)
+    {
+      return getPlugin(handle);
+    }
+  }
+  return nullptr;
+}
```

Comment:
I would expect getPlugin to use plugin name, not filter names... There is no method documentation saying that the plugin name has nothing to do with getPlugin(const std::string&), so I can only assume that this is incorrect or at best misleading. Either add method documentation explaining what the method does and why it only uses filter names (still not a good idea without collision checking) or update the method to do what it's name implies as that is the only thing currently explaining what the method does.

There's also far less plugins than FilterHandles, so if you are looking for a plugin by name, it would be better to iterate on those.

Reply:
I removed this API.